### PR TITLE
[Android] Border Stroke gradient can only switch to another gradient - fix

### LINKED ISSUE #22042: Android : Border Stroke GradiantBrush can only switch to another gradiantbrush
### Description

Hi, 
I just saw that when i set the Stroke of a Border with a gradiant resources, i can't change it back to a classic color or even a solidcolorbrush. For exemple switch from a gradiant to a gray color don't work and the gradiant stay.

So i tried by setting a fake gradiant with 2 stops of the same color (gray) and it worked.

Tried on Android only but the problem could exist on other platforms, dunno

### Steps to Reproduce

Add GradiantBrush to Colors.xaml
Add a border to your xaml with a tapgesture.
Switch from a color to another.

Hope you'll find a way to fix that !

Also... if you have plan to add gradiant on textcolor, it could be nice :p
This is my first time reporting a bug, hope i've done it right :)

### Link to public reproduction project repository

https://github.com/Skarzag/MauiAppGradiantBorder

### Version with bug

8.0.21 SR4.1

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by kubaflo**
### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/22042

|Before|After|
|--|--|
|<video src="https://github.com/dotnet/maui/assets/42434498/e81549e7-319a-4f4c-82a9-b032fd3f3756" width="300px"/>|<video src="https://github.com/dotnet/maui/assets/42434498/928c234d-3530-469b-8dbf-97758462e019" width="300px"/>|

### CODE REVIEW

**Review by jsuarezruiz — DISMISSED** [review]

File: src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22042.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		{
+		}
+
+		public override string Issue => "[Android] Border Stroke GradiantBrush can only switch to another gradiantbrush";
+
+		[Test]
+		[Category(UITestCategories.Border)]
+		public void BorderCollorShouldChange()
+		{
+			this.IgnoreIfPlatforms(new TestDevice[] { TestDevice.iOS, TestDevice.Mac, TestDevice.Windows });
```
Comment (PureWeen) [inline src/Controls/tests/TestCases.Shared.Tests/Tests/Issues/Issue22042.cs:19]: Can you update this to use if/defs instead of "this.IgnoreIfPlatforms"

**Review by PureWeen — CHANGES_REQUESTED** [review]

**Review by PureWeen — APPROVED** [review]

**Pull request merged**